lib/google_contacts_api/group.rb

```diff
@@ -40,5 +40,18 @@ def edit_link
       _link = self["link"].find { |l| l.rel == "edit" }
       _link ? _link.href : nil
     end
+
+    def self.create(attrs, api)
+      response = api.post('groups/default/full', xml_for_create(attrs), {}, 'Content-Type' => 'application/atom+xml')
+      Group.new(Hashie::Mash.new(JSON.parse(response.body)).entry)
+    end
+
+    def self.xml_for_create(group_attrs)
+      @@new_group_template ||=
+        File.new(File.dirname(__FILE__) + '/templates/group.xml.erb').read
+
+      # Passes in group_attrs to the template via binding
+      ERB.new(@@new_group_template).result(binding)
```

Comment:
Yep, you're right, it's stored in the class variable. But still, having a separate template is not correct. I think the whole group creation should be considered with the contact creation: these are actually two very similar features, that share the same logic. The only differences between them are the url and parameters. That's why it's not good to store a template for groups. It's just a one-time solution. Instead, we need to think about a mechanism that accepts parameters for a new contact or group and then generates an XML using these parameters. We need to have one mechanism to create entities, that will be shared by contact and group creation.
If you agree with this I'll later create an issue with general design of such mechanism.